pkg/sriov/sriov.go

```diff
@@ -0,0 +1,113 @@
+//
+// Use and distribution licensed under the Apache license version 2.
+//
+// See the COPYING file in the root project directory for full text.
+//
+
+package sriov
+
+import (
+	"fmt"
+
+	"github.com/jaypipes/ghw/pkg/context"
+	"github.com/jaypipes/ghw/pkg/marshal"
+	"github.com/jaypipes/ghw/pkg/option"
+	"github.com/jaypipes/ghw/pkg/pci"
+	pciaddr "github.com/jaypipes/ghw/pkg/pci/address"
+)
+
+type Device struct {
+	Interfaces []string `json:"interfaces"`
+	// the PCI address where the SRIOV instance can be found
+	Address *pciaddr.Address `json:"address"`
+	PCI     *pci.Device      `json:"pci"`
+}
+
+func (d Device) ToString(devType string) string {
+	deviceStr := d.Address.String()
+	nodeStr := ""
+	if d.PCI != nil {
+		deviceStr = d.PCI.String()
+		if d.PCI.Node != nil {
+			nodeStr = fmt.Sprintf(" [affined to NUMA node %d]", d.PCI.Node.ID)
+		}
+	}
+	return fmt.Sprintf("%s function %s@%s", devType, nodeStr, deviceStr)
+}
+
+type PhysicalFunction struct {
+	Device
+	MaxVFNum int               `json:"max_vf_num,omitempty"`
+	VFs      []VirtualFunction `json:"vfs,omitempty"`
+}
+
+type VirtualFunction struct {
+	Device
+	ID int `json:"id"`
```

Comment:
Is this the index of the VF within the PF? If so, should we call this `Index`?

Reply:
Yep, it should be called probably `Index` and I should clarify it with a code comment.